### ISSUE TITLE
Enabling "either()" for hash keys

### ISSUE THREAD

**Issue by SherModo** [tomdalling/rschema#11]
It would be useful to enable the use of "either" for hash keys.

For example, instead of having to do this:

```
either(
        variable_hash(
          _Symbol => fixed_hash(:foo => _Integer, :bar => _String)
        ),
        variable_hash(
          _String => fixed_hash("foo" => _Integer, "bar" => _String)
        )
      )
```

something like this would be cleaner and more compact:

```
variable_hash(
        either(_Symbol, _String) => fixed_hash(
          either(:foo, 'foo') => _Integer,
          either(:bar, 'bar') => _String
        )
      )
```

If there is a way to accept a symbol or its equivalent string as a key, that would also solve the above issue. I tried using `enum([:foo, 'foo'])`, but it didn't work.

**Comment by tomdalling** [comment]
Before getting into other approaches, would coercion work for your use case? For example, the result would always contain symbols, but it can accept either symbols or strings as input:

```ruby
# coercion wrappers are configurable, but i'm just using RACK_PARAMS
# because it already exists in the RSchema gem
schema = RSchema::CoercionWrapper::RACK_PARAMS.wrap(
  RSchema.define do
    variable_hash(_Symbol => fixed_hash(
      foo: _Integer,
      bar: _Integer,
    ))
  end
)

string_input = { "apple" => { "foo" => 1, "bar" => 2 } }
symbol_input = { apple: { foo: 1, bar: 2 } }

result1 = schema.validate!(string_keys)
result2 = schema.validate!(symbol_keys)

result1 == result2 == symbol_input #=> true
```

The reason why your example code doesn't work is because fixed hashes expect the keys to be the actual keys found in the hash, not a schema object (like an enum). It should already work for variable hashes though, because they _do_ take a schema object for the keys.

**Comment by SherModo** [comment]
Thank you for the prompt response!

Coercion can be used for my use case. I tested the snippet you suggested and it worked.

Regarding variable hashes and using schema objects for their keys, you're right, I think I used it at some point during the testing, but overlooked it when submitting the issue.

On a different note, it would be useful to add a section about installing the gem in Rails using:

`gem 'rschema', require: 'rschema/rails'`

Thanks a lot!
